app/assets/javascripts/discourse/app/lib/screen-track.js

```diff
@@ -85,6 +89,99 @@ export default class {
     this._anonCallback = cb;
   }
 
+  consolidateTimings(timings, topicTime, topicId) {
+    this._consolidatedTimings = this._consolidatedTimings || [];
+
+    let found = this._consolidatedTimings.find((elem) => elem[2] === topicId);
+
+    if (found) {
+      this._consolidatedTimings.removeObject(found);
```

Comment:
Why do we remove the object only to add it again below? (`unshift`)

Removing an object in JS is way slower than modifying it in place. It would be less work to not remove it, and only unshift a record that was not found in the `else` clause.

Reply:
Yeah I changed it so we use push/pop instead so freshest is at the end. The means we lean now on findIndex which is not going to work on ie11.